### PR TITLE
use Pm_GetErrorText for errors, export ErrMinBuffer and ErrMaxBuffer, …

### PR DESCRIPTION
**Pull request by briansorahan**
… and return errors from Stream factories if stream is unavailable

### CODE REVIEW

File: portmidi.go [hunk 1]
```diff
@@ -111,10 +86,7 @@ func Time() Timestamp {
...
 }
 
+// convertToError converts a portmidi error code to a Go error.
 func convertToError(code C.PmError) error {
-	err, ok := errorMap[int(code)]
-	if !ok && code != 0 {
-		return ErrUnknown
-	}
-	return err
+	return errors.New(C.GoString(C.Pm_GetErrorText(code)))
```
Comment (rakyll) [inline portmidi.go]: Are we leaking here? Once we convert the C string to Go string, should we deallocate?

File: portmidi.go [hunk 2]
```diff
@@ -25,35 +25,10 @@ import (
...
-	3:      ErrBufferTooSmall,
-	4:      ErrBufferOverflow,
-	5:      ErrBadPtr,
-	6:      ErrBadData,
-	7:      ErrInternalError,
-	8:      ErrBufferMaxSize,
-}
-
+// DeviceId is a MIDI device ID.
 type DeviceId int
```
Comment (rakyll) [inline portmidi.go]: type DeviceID int

File: portmidi.go [hunk 3]
```diff
@@ -25,35 +25,10 @@ import (
...
 
+// DeviceInfo provides info about a MIDI device.
 type DeviceInfo struct {
 	Interface         string
```
Comment (rakyll) [inline portmidi.go]: What does this field represent?

File: portmidi.go (hunk 1 above)
Comment (briansorahan) [inline portmidi.go]: The C string is a compile-time constant, so I don't think so.

File: portmidi.go (hunk 3 above)
Comment (briansorahan) [inline portmidi.go]: Not too sure, but I think it's the underlying lib that portmidi uses, like "ALSA" or "CoreMIDI" or something like that (judging by [this](https://sourceforge.net/p/portmedia/code/HEAD/tree/portmidi/trunk/pm_common/portmidi.h#l209))

File: portmidi.go [hunk 4]
```diff
@@ -25,35 +25,10 @@ import (
...
-	3:      ErrBufferTooSmall,
-	4:      ErrBufferOverflow,
-	5:      ErrBadPtr,
-	6:      ErrBadData,
-	7:      ErrInternalError,
-	8:      ErrBufferMaxSize,
-}
-
-type DeviceId int
+// DeviceId is a MIDI device ID.
```
Comment (kisielk) [inline portmidi.go]: Docs need to be fixed here

**Comment by rakyll** [comment]
LGTM, @briansorahan, have you signed a Google CLA before? I cannot find your name in the records. If not, could you please take a look at https://cla.developers.google.com/clas?

**Comment by briansorahan** [comment]
Done -_-

**Comment by rakyll** [comment]
Thanks!

**Pull request merged**